### PR TITLE
feat: Anchor support cssVar

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

- #45618

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Anchor support cssVar |
| 🇨🇳 Chinese | feat: Anchor 组件支持 cssVar |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at f103ed7</samp>

This pull request adds support for CSS variables to the `Anchor` and `AnchorLink` components, which enables more flexible and dynamic styling. It also adds a `displayName` property to the `AnchorLink` component for debugging and testing. The main changes are in the files `components/anchor/Anchor.tsx`, `components/anchor/style/index.ts`, `components/anchor/AnchorLink.tsx`, and `components/anchor/style/cssVar.ts`.

### <samp>🤖 Generated by Copilot at f103ed7</samp>

*  Import and use `useCSSVar` hook to register CSS variables for anchor component ([link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-353746edad3f07acb175d7e32c64972b87359b5a0dc4e14e4d3f1133b61e48a8R16), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-353746edad3f07acb175d7e32c64972b87359b5a0dc4e14e4d3f1133b61e48a8L368-R372))
*  Add `displayName` property to `AnchorLink` component for debugging and testing ([link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-584bab7d33a5231e7e6e844bdb79bf36b1b1abd13e03d1f45eaa3bf03d0f6ce7R94-R97))
*  Create and export `genCSSVarRegister` function to generate `useCSSVar` hook for anchor component in `cssVar.ts` file ([link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-440023592e3aaf4f154af53432786b63ab5d4b84230967b2365e1520bdff4952R1-R4))
*  Import and apply `unit` function from `@ant-design/cssinjs` to convert token values to valid CSS units in `index.ts` file ([link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL2-R5), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL11-R25), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL40-R43), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL49-R55), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL59-R61), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL74-R76), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL86-R88), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL96-R99), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL153-R154), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL161-R177))
*  Extract and export `prepareComponentToken` function to prepare default token values for CSS variables in `index.ts` file and pass it to `genCSSVarRegister` function in `cssVar.ts` file ([link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL161-R177), [link](https://github.com/ant-design/ant-design/pull/45808/files?diff=unified&w=0#diff-440023592e3aaf4f154af53432786b63ab5d4b84230967b2365e1520bdff4952R1-R4))

### CODE REVIEW

File: components/anchor/AnchorLink.tsx [hunk 1]
```diff
@@ -91,4 +91,8 @@ const AnchorLink: React.FC<AnchorLinkProps> = (props) => {
   );
 };
 
+if (process.env.NODE_ENV !== 'production') {
+  AnchorLink.displayName = 'AnchorLink';
+}
```
Comment (MadCcc) [inline components/anchor/AnchorLink.tsx:96]: 这个不用加

File: components/anchor/style/index.ts [hunk 2]
```diff
@@ -33,30 +35,31 @@ const genSharedAnchorStyle: GenerateStyle<AnchorToken> = (token): CSSObject => {
...
+      paddingBlockStart: unit(holderOffsetBlock),
 
       // delete overflow: auto
       // overflow: 'auto',
 
       [componentCls]: {
         ...resetComponent(token),
         position: 'relative',
-        paddingInlineStart: lineWidthBold,
+        paddingInlineStart: unit(lineWidthBold),
```
Comment (MadCcc) [inline components/anchor/style/index.ts:52]: 只要替换有 px 的就行了，这种就都不用改

File: components/anchor/style/index.ts [hunk 3]
```diff
@@ -128,7 +130,7 @@ const genSharedAnchorHorizontalStyle: GenerateStyle<AnchorToken> = (token): CSSO
           value: 0,
         },
         bottom: 0,
-        borderBottom: `1px ${token.lineType} ${token.colorSplit}`,
+        borderBottom: `${unit(1)} ${token.lineType} ${token.colorSplit}`,
```
Comment (MadCcc) [inline components/anchor/style/index.ts:133]:
```suggestion
        borderBottom: `${unit(token.lineWidth)} ${token.lineType} ${token.colorSplit}`,
```

File: components/anchor/style/index.ts [hunk 4]
```diff
@@ -150,20 +152,24 @@ const genSharedAnchorHorizontalStyle: GenerateStyle<AnchorToken> = (token): CSSO
...
+
 // ============================== Export ==============================
 export default genComponentStyleHook(
   'Anchor',
   (token) => {
     const { fontSize, fontSizeLG, paddingXXS } = token;
-
     const anchorToken = mergeToken<AnchorToken>(token, {
       holderOffsetBlock: paddingXXS,
       anchorPaddingBlockSecondary: paddingXXS / 2,
```
Comment (MadCcc) [inline components/anchor/style/index.ts:175]: 这个要改

**Comment by MadCcc** [comment]
image
demo 都有些问题。
`/scripts/check-cssinjs.tsx` 里把 ignore 里的 anchor 去掉，可以本地跑 lint

File: components/anchor/Anchor.tsx [hunk 5]
```diff
@@ -365,9 +366,10 @@ const Anchor: React.FC<AnchorProps> = (props) => {
   const { getPrefixCls } = React.useContext<ConfigConsumerProps>(ConfigContext);
   const anchorPrefixCls = getPrefixCls('anchor', customizePrefixCls);
 
-  const [wrapSSR, hashId] = useStyle(anchorPrefixCls);
+  const [, hashId] = useStyle(anchorPrefixCls);
+  const wrapCSSVar = useCSSVar(anchorPrefixCls);
```
Comment (MadCcc) [inline components/anchor/Anchor.tsx:370]: 需要用 useCSSVarCls 生成 rootCls 给到 rootClassName，Anchor 的最外层不是`ant-anchor`，现在组件 token 都没生效

Comment (li-jia-nan) [reply components/anchor/Anchor.tsx:370]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**